### PR TITLE
cli: retry failed deletes

### PR DESCRIPTION
**Pull request by ironcladlou**
When deleting a cluster or infrastructure, retry failed deletes until
an interrupt is received. This helps backstop transient issues with
internal retry handling (e.g. AWS dependencies which need cleaned up).

### CODE REVIEW

File: cmd/infra/aws/destroy.go [hunk 1]
```diff
@@ -53,14 +54,23 @@ func NewDestroyCommand() *cobra.Command {
...
+		t := time.NewTicker(5 * time.Second)
+		for {
+			select {
+			case <-ctx.Done():
+				log.Info("Destroy was cancelled")
+				return nil
+			case <-t.C:
+				err := opts.DestroyInfra(ctx)
+				if err != nil {
+					log.Error(err, "failed to destroy infrastructure, will retry")
```
Comment (enxebre) [inline cmd/infra/aws/destroy.go:66]: nit: can inline this

**Comment by enxebre** [comment]
small nit, lgtm

File: cmd/infra/aws/destroy.go (hunk 1 above)
Comment (ironcladlou) [reply cmd/infra/aws/destroy.go:65]: Fixed

**Pull request merged**